Strategy/IFileReader.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WeatherMonitoringSystem.model;
+
+namespace WeatherMonitoringSystem.Strategy
+{
+    public interface IFileReader
```

Comment:
Here we're not reading a file, we're just deserializing a string.
```suggestion
    public interface ISerializer
```